projects/core/src/lib/colors/color-map-selector/color-map-selector.component.ts

```diff
@@ -211,17 +211,36 @@ export class ColorMapSelectorComponent implements OnInit, OnDestroy, OnChanges {
     ): Array<ColorBreakpoint> {
         const breakpoints = new Array<ColorBreakpoint>();
 
-        const stepSize = 1 / (colorMapSteps - 1);
-        for (let frac = 0; frac <= 1; frac += stepSize) {
+        for (let i = 0; i <= colorMapSteps; i++) {
+            const frac = i / colorMapSteps;
             const value = bounds.min + frac * (bounds.max - bounds.min);
 
-            let colorMapIndex = Math.round(frac * (colorMap.length - 1));
+            const colorMapPos = frac * (colorMap.length - 1);
+            let colorMapIndex = Math.floor(colorMapPos);
+            const colorMapPosRemainder = colorMapPos - colorMapIndex;
+            let nextMapIndex = Math.min(colorMapIndex + 1, colorMap.length - 1);
+
             if (colorMapReverseColors) {
                 colorMapIndex = colorMap.length - colorMapIndex - 1;
+                nextMapIndex = Math.max(0, colorMapIndex - 1);
+            }
+
+            if (colorMapPosRemainder < Number.EPSILON) {
+                breakpoints.push(new ColorBreakpoint(value, Color.fromRgbaLike(colorMap[colorMapIndex])));
+                continue;
             }
+
             const color = Color.fromRgbaLike(colorMap[colorMapIndex]);
+            const nextColor = Color.fromRgbaLike(colorMap[nextMapIndex]);
 
```

Comment:
putting this comment here randomly:

how can I try this out with the default datasets? I tried using it on land cover and ndvi but they are both palettes and if I change from palette to linear colorizer nothing, create the breakpoints and apply them, nothing happens.

Reply:
there was this issue with the float rgba values. I fixed that: [d31466b](https://github.com/geo-engine/geoengine-ui/pull/680/commits/d31466ba837ef2fb26d0b6740bdfbc7b47c3cf37)

Also i fixed the switch between linear and log scale: [df92651](https://github.com/geo-engine/geoengine-ui/pull/680/commits/df926516b3d3cf98d814bbc58f984ea50d1d5e7d)

I always use the NDVI3857 dataset. It does not use a palette.

Reply:
![image](https://github.com/geo-engine/geoengine-ui/assets/10283892/8863f444-3b0b-4476-a8de-5badfff53f2c)

I tried it out, but if I select 2 steps it produces 3. Do you want to fix this here real quick or shall I create a new card?

Everything else seems to work.

Reply:
[d5303fd](https://github.com/geo-engine/geoengine-ui/pull/680/commits/d5303fd5d104632bba8edb5c3b4d80f06e3d0169)